kazoo/tests/test_utils.py

```diff
@@ -0,0 +1,22 @@
+import unittest
+
+from mock import patch
+
+
+class TestCreateTCPConnection(unittest.TestCase):
+    def test_timeout_arg(self):
```

Comment:
It seems to me that this test is currently calling the `create_connection` method with a timeout of `0.5`. Should we add a test to verify that when using `create_tcp_connection(socket, ('127.0.0.1', 2181), timeout=0.5)` (that means, `timeout_at` is < 0), a `socket.error` is raised? Maybe that would also be better to change this `socket.error` to something more specific.